Firestore/core/src/firebase/firestore/model/document.h

```diff
@@ -59,67 +61,64 @@ std::ostream& operator<<(std::ostream& os, DocumentState state);
  */
 class Document : public MaybeDocument {
  public:
+  Document() = default;
+
   /**
    * Construct a document. ObjectValue must be passed by rvalue.
    */
-  Document(ObjectValue&& data,
+  Document(ObjectValue data,
            DocumentKey key,
            SnapshotVersion version,
            DocumentState document_state);
 
+  /**
+   * Construct a document. ObjectValue must be passed by rvalue.
+   */
+  Document(ObjectValue data,
+           DocumentKey key,
+           SnapshotVersion version,
+           DocumentState document_state,
+           absl::any proto);
+
+  /**
+   * Casts a MaybeDocument to a Document. This is a checked operation that will
+   * assert if the type of the MaybeDocument isn't actually Type::Document.
+   */
+  explicit Document(const MaybeDocument& document);
```

Comment:
There's a bunch of places where the `MaybeDocument` could be moved into the `Document`, but given that copying it is cheap, it's probably not worth the trouble.

Reply:
I think [here](https://github.com/firebase/firebase-ios-sdk/pull/3531/files#diff-df205eaa49ddf9029c664b8a454f7cceR364) is an example (I should have mentioned they are all in the follow-up PR).